juju/juju
# Clear tech debt by migrating remaining uses of application offer tags to use offer uuids.

Application offer tags were updated to use the uuid not the name. However, for compatibility reasons, some places still relied on the name, specifically the remote entity token lookups used to manage cross model relation exchanges. So the codebase had compatibility shims in various places.

This PR updates the code to only use offer uuids. For dealing with existing data, it needs an upgrade step and model migration enhancements to convert tags in the database to the correct format.

## QA steps

There's 3 scenarios: new cmr deployment, upgrade existing cmr deployment, migrate cmr model.

cmr deployment can be done by: 
deploy `juju-qa-dummy-source` and offering `dummy-source:sink`
in another model, deploy `juju-qa-dummy-sink` and relate to the dummy-source offer
in the dummy-source app, set `token` config and status will show all green

for upgrade test, do the cmr deployment but don't set token
upgrade controller
set token and ensure status shows green on both models

for migrate test, do the cmr deployment but don't set token
migrate the offering model to a new controller
set the token
both models should have green status

after upgrade/migration, look at the content of the `remoteEntities` collection and see records with `applicationoffer-<uuid>`

## Links

**Jira card:** JUJU-5204
